clinvar_ingest/api/model/requests.py

```diff
@@ -108,14 +109,59 @@ def _validate(cls, v):  # pylint: disable=E0213
         return v
 
 
+class GcsBucketName(RootModel):
+    root: str
+
+    @validator("root")
+    def _validate(cls, v):  # pylint: disable=E0213
+        alpha_lower = "abcdefghijklmnopqrstuvwxyz"
+        alpha_upper = alpha_lower.upper()
+        nums = "0123456789"
+        symbols = "-_."
+        allowed_chars = set(alpha_lower + alpha_upper + nums + symbols)
+
+        for c in v:
+            if c not in allowed_chars:
```

Comment:
@theferrit32 I'm curious if this is a better (faster and easier to maintain) approach vs using a regex?

Reply:
There's probably not much difference between this and doing a regex check with `[a-zA-Z0-9-_.]`. Was just testing with this, decided against using it, I'll delete this class. If the bucket name is invalid the google API will eventually throw an error.